src/CodeGen_OpenCL_Dev.cpp

```diff
@@ -452,7 +452,7 @@ void CodeGen_OpenCL_Dev::CodeGen_OpenCL_C::add_kernel(Stmt s,
     for (size_t i = 0; i < args.size(); i++) {
         if (args[i].is_buffer()) {
             stream << " " << get_memory_space(args[i].name) << " ";
-            if (!args[i].write) stream << "const ";
+            if (!args[i].is_output()) stream << "const ";
```

Comment:
Looks like a search-replace. I think if (args[i].is_input()) stream << "const " would be better.